### PR TITLE
add algorithm-option, set default hash algorithm to sha256

### PR DESCRIPTION
**Pull request by Uzlopak**
Motivated by the latest csurf sec. vuln. 

snyk writes regarding csurf:

> In particular, it doesn’t properly validate cookie values for the double submit cookie pattern, which results in attackers being able to spoof the CSRF token. Adding to the problem, it relies on a deprecated encryption algorithm (SHA1) for hashing CSRF tokens.

https://snyk.io/blog/explaining-the-csurf-vulnerability-csrf-attacks-on-all-versions/

I personally think sha1 or sha256 or whatever does not make a big difference as csrf is actually shortlived so there is no time to crack it actually. But this PR adds anyway the algorithm option. 

It sets sha256 as default hash algorithm. It would be a breaking change, but people could upgrade and set the algorithm back to sha1 if they still want to use sha1.

It this gets merged, there would be a follow up PR for the csrf-protection package.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
@climba03003 
@mcollina 
@Eomm 
@jsumners

**Review by mcollina — APPROVED** [review]
I'm ok with making the default sha256

**Comment by Uzlopak** [comment]
@mcollina 

Changed to use sha256 by defaultl

File: README.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ optional and will just use all defaults if missing.
 
 Tokens accepts these properties in the options object.
 
+##### algorithm
+
+The hash-algorithm to generate the token. Defaults to `sha1`.
```
Comment (Eomm) [inline README.md:53]:
```suggestion
The hash-algorithm to generate the token. Defaults to `sha256`.
```

File: types/index.d.ts [hunk 2]
```diff
@@ -38,6 +38,12 @@ interface TokensUserinfo extends TokensBase {
 export type SecretCallback = (err: Error | null, secret: string) => void;
 
 export interface Options {
+  /**
+   * The algorithm used to generate the token
+   * @default sha1
```
Comment (Eomm) [inline types/index.d.ts:43]:
```suggestion
   * @default sha256
```

**Review by Eomm — APPROVED** [review]

**Pull request merged**